movieApp/movieSeriesListViewController.swift

```diff
@@ -0,0 +1,118 @@
+//
+//  movieSeriesListViewController.swift
+//  movieApp
+//
+//  Created by <3 on 18.05.2024..
+//
+
+import Foundation
+import UIKit
+import PureLayout
+import MovieAppData
+
+class AllMoviesViewController: UIViewController, MovieNavigationDelegate {
```

Comment:
You can create AllMoviesViewController+Design.swift file to store all the design logic in it. It would look something like this:

```
extension AllMoviesViewController {

// logic to init the views
func createViews() {
    collectionView = UICollectionView(frame: .zero, collectionViewLayout: layout)
    collectionView.register(AllMoviesSectionCell.self, forCellWithReuseIdentifier: "MovieCollectionViewCell")
    view.addSubview(collectionView)
}

func styleViews() {
      collectionView.delegate = self
      collectionView.dataSource = self
      collectionView.backgroundColor = .white
}

func defineLayoutForViews() {
    collectionView.autoPinEdgesToSuperviewEdges()
}

}
```

Reply:
This approach ensures your ViewController file contains real logic, and +Design file makes sure that everything is styled correctly.

Additionally, this approach leaves `collectionView.reloadData()` out because that doesn't actually belong in any of the 3 stated funcs. That's a good indicator it should be placed somewhere else. Perhaps in 

```
override func viewWillAppear(_ animated: Bool) {
    super.viewWillAppear(animated)    

    collectionView.reloadData()
}
```


As you can see, separating code into designated functions can increase code clarity and also ensure you don't have faulty behaviours. It also helps you identify bugs much easier